.teamcity/src/main/kotlin/projects/CheckProject.kt

```diff
@@ -19,7 +19,10 @@ class CheckProject(
 
     params {
         param("credentialsStorageType", "credentialsJSON")
+        // Disallow Web UI changes to TeamCity settings
         param("teamcity.ui.settings.readOnly", "true")
+        // Avoid rebuilding same revision if it's already built on another branch (pre-tested commit)
+        param("teamcity.vcsTrigger.runBuildOnSameRevisionInEveryBranch", "false")
```

Comment:
Let's hope this works. Are you going to merge this now to try it out, or can we try out the change without merging it?

Reply:
Confirmed to be working with an experimental pipeline.